client/a8c-for-agencies/sections/marketplace/assign-license/index.tsx

```diff
@@ -104,23 +115,46 @@ export default function AssignLicense( { sites, currentPage, search }: Props ) {
 
 	const onClickAssignLater = useCallback( () => {
 		if ( licenseKeysArray.length > 1 ) {
-			page.redirect( '/partner-portal/licenses' );
+			page.redirect( A4A_LICENSES_LINK );
 		}
 
 		const licenseKey = licenseKeysArray[ 0 ];
-		page.redirect( addQueryArgs( { highlight: licenseKey }, '/purchases/licenses' ) );
+		page.redirect( addQueryArgs( { highlight: licenseKey }, A4A_LICENSES_LINK ) );
 	}, [ licenseKeysArray ] );
 
-	const onClickAssignLicenses = useCallback( () => {
+	const onClickAssignLicenses = useCallback( async () => {
 		dispatch(
 			recordTracksEvent( 'calypso_a4a_assign_multiple_licenses_submit', {
 				products: licenseKeysArray.join( ',' ),
 				selected_site: selectedSite?.ID,
 			} )
 		);
 
-		// TODO: Need to implement actual assign Licenses operation once we have a working API
-	}, [ dispatch, licenseKeysArray, selectedSite?.ID ] );
+		const assignLicensesResult = await assignLicensesToSite( licenseKeysArray );
+
+		dispatch( resetSite() );
+		dispatch( setPurchasedLicense( assignLicensesResult ) );
+
+		const goToDownloadStep = licenseKeysArray.some( ( licenseKey ) =>
+			isWooCommerceProduct( licenseKey )
+		);
+
+		if ( goToDownloadStep ) {
+			return page.redirect(
+				addQueryArgs(
+					{ products: licenseKeysArray.join( ',' ), attachedSiteId: selectedSite?.ID },
+					A4A_DOWNLOAD_PRODUCTS_LINK
+				)
+			);
+		}
+
+		const fromDashboard = getQueryArg( window.location.href, 'source' ) === 'dashboard';
+		if ( fromDashboard ) {
+			return page.redirect( '/sites' );
```

Comment:
We can also use A4A_SITES_LINK here

Reply:
Sorry for missing this earlier. Changes done: https://github.com/Automattic/wp-calypso/pull/88504/commits/d63e51546de1a93a298748beedd0772094720708